web-client/public/js/update-app.js

```diff
@@ -512,6 +512,15 @@ const toggleLayout = (on, off) => {
     }
 };
 
+const hasExpressionData = (sheets) => {
+    for (var property in sheets) {
+        if (property.match(ENDS_IN_EXPRESSION_REGEXP)) {
+            return true;
+        }
+    }
+    return false;
+};
```

Comment:
This isn’t functionally incorrect, but along the lines of more current idioms when processing lists or objects, see if this can be written as a `some` call? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Using `some` will make the code look something like this…

```suggestion
const hasExpressionData = (sheets) => {
    return (/*expression with sheets*/).some(
        value => (/*expression with value */).match(ENDS_IN_EXPRESSION_REGEXP)
    )
};
```

…which to me sounds more concise, and objectively does not use side effects as well as easier for an optimizer to parallelize

I realize `sheets` isn’t an array, but there may be ways to take advantage of the simpler expressiveness of `some` vs. having a `return true` with a `return false` at the end (that’s why I have some placeholder `/* */` comments in there instead of specific code; there may be other places too)

Reply:
I just modified it to use `some` call instead.

Reply:
You certainly modified it in _graph.js_, but does that mean that lines 515–522 above are duplicated?